enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,483 @@
+configured to a different metrics collection profile (`full`, `minimal`). Then
+we would trigger a workload using
+[kube-burner](https://github.com/cloud-bulldozer/kube-burner) and at the end of
+2 hours, we evaluated the results. We concluded that in terms of resource usage,
+given the results obtained we can confidently state that the feature is quite
+valuable given the reduction of CPU by ~21% and memory by ~33% when comparing
+the `minimal` to the `full` profile. More details can be consulted in this
+[document](https://docs.google.com/document/d/1MA-HTJQ_X7y_bwpJS2IPbGmC4qDyIMp25jisr34X2F4/edit?usp=sharing).
+
+Moreover, through Telemetry, we collect for each cluster the top 3 Prometheus
+jobs that generate the biggest amount of samples. With this data, we know that
+for OpenShift 4.11 the 5 components most often reported as the biggest producers
+are: the Kubernetes API servers, the Kubernetes schedulers, kube-state-metrics,
+kubelet and the network daemon.
+
+
+### User Stories
+
+- As a user, I want to lower the amount of resources consumed by Prometheus in a
+  supported way, so I can configure the clusters metrics collection profiles to
+  `minimal`.
+- As a developer, I want a supported way to collect a subset of the metrics
+  exported by my operator and operands, while still collecting necessary metrics
+  for alerts, visualization of key indicators and Telemetry.
+- As a developer of a component (that does not yet implement a profile), I want
+  to extract metrics needed to implement said profile, based on the assets I
+  provide, or the ones gathered from the cluster based on a group of target
+  selectors, and a plug-in relabel configuration to apply within the monitor.
+- As a developer of a component (that does not, or only partially implements a
+  profile), I want to get information about any monitors that are not yet
+  implemented for any of the supported profiles that are offered.
+- As a developer of a component (that implements a profile), I want to verify if
+  all the profile metrics are present in the cluster, and which of the profile
+  monitors are affected if not. Also, I want additional information to narrow
+  down where these metrics are exactly being used.
+
+### Goals
+
+- Give users a supported and documented method to pick the amount of metrics
+  being collected by the platform, in a way that allows
+  cluster-monitoring-operator and other operators to provide their own
+  defaulting.
+- Other components of the cluster (alerts, console, HPA and VPA and telemetry)
+  should not be negatively affected by this feature.
+- Developers may optionally adhere to this feature, those that do not adhere
+  should not be impacted.
+
+### Non-Goals
+
+- Dynamically adjusting metrics scraped at runtime based on heuristics.
+- Allowing users to explicitly set a list or a regex of metrics that they would
+want to be included in a given profile.
+- Support the addition of profiles other than `full` and `minimal`.
```

Comment:
Assuming the Collection Profiles operate in a decoupled manner w.r.t. installer, (1) is something we have anticipated, and have support for within the current framework. The workflow would go about like so:

- CMO introduces a `custom` profile.
- Teams implement said profile on their end, [like so](assets/kube-state-metrics/minimal-service-monitor.yaml).
- Teams configure their respective component to detect the currently configured profile and deploy the appropriate resources needed to sustain that profile.

_I believe the EP is outdated to reflect certain advancements we have made recently. I make sure these are reflected here soon, since we now have answers for [initially proposed questions](https://github.com/openshift/enhancements/pull/1298/files#diff-d53514a51b0340df4d66642f8263b41c912d81fdb8f87522fefd4f318a9e3511R341-R344)._

(2) is something that's not addressed in the current implementation, but definitely something that's good to have. While we are indeed leaning towards keeping things simple from both sides and thus onboard and [help as many teams leverage everything the featureset currently has to offer](https://github.com/rexagod/cpv), I'll bring this up with the team and update the EP with planning to implement this in the later iterations.